dmscripts/mark_definite_framework_results.py

```diff
@@ -83,12 +83,21 @@ def mark_definite_framework_results(
     dry_run=True,
     supplier_ids=None,
     logger=logging.getLogger("script"),
+    excluded_supplier_ids=None,
 ):
     interested_supplier_ids = supplier_ids or client.get_interested_suppliers(
         framework_slug
     ).get('interestedSuppliers', ())
     total_interested_suppliers = len(interested_supplier_ids)
 
+    # exclude suppliers with IDs the executioner defined
+    if excluded_supplier_ids is not None:
+        for excluded_supplier in excluded_supplier_ids.split(','):
+            try:
+                interested_supplier_ids.pop(interested_supplier_ids.index(int(excluded_supplier)))
+            except IndexError:  # occurs when .index() fails
+                continue
```

Comment:
Y'know we could just make both `interested_supplier_ids` and `excluded_supplier_ids` `set`s and do set operations here instead of faffy index lookups. It's not like we really care about the ordering is it?

Reply:
Yes, `interested_supplier_ids = list(set(interested_supplier_ids) - set(excluded_supplier_ids))`, but this would require knowledge of sets.

Reply:
Well, we don't even need to convert them back to a list afterwards, and it's fine to use sets - we're all adults here. If we want to make it more explicit we can always use `set`'s named methods rather than the operators...